Core/src/Batch/HandleFailureTrait.php

```diff
@@ -88,7 +88,7 @@ public function handleFailure($idNum, array $items)
 
         if ($this->failureFile) {
             $fp = @fopen($this->failureFile, 'a');
-            @fwrite($fp, serialize([$idNum => $items]) . PHP_EOL);
+            @fwrite($fp, json_encode(serialize([$idNum => $items])) . PHP_EOL);
```

Comment:
Can we just call `trim()` here?

Reply:
If I understand your idea correctly, you're suggesting to remove leading and trailing whitespaces. If this is the case then...our error case is little different.

If any job item (let's say `"Hello World"`) contains the end of line character even in the middle of it (like this: `"Hello" . PHP_EOL . "World"`), then also it would throw error as `serialize()` method doesn't escapes those characters making is line reading proof.

Moreover, data is something the user has and wants it to be processed as it is, removing intended whitespaces from either end would be incorrect functionality.

Reply:
understood, I assumed that the EOL char is only towards the end